app/src/main/java/com/jalasoft/routesapp/ui/tourPoints/viewModel/TourPointsViewModel.kt

```diff
@@ -1,26 +1,23 @@
 package com.jalasoft.routesapp.ui.tourPoints.viewModel
 
-import android.content.Context
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
+import com.jalasoft.routesapp.data.local.room.interfaces.TourPointLocalRepository
 import com.jalasoft.routesapp.data.model.remote.TourPointPath
-import com.jalasoft.routesapp.data.remote.interfaces.TourPointRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class TourPointsViewModel
 @Inject
-constructor(private val repository: TourPointRepository) : ViewModel() {
+constructor(private val tourPointLocalDB: TourPointLocalRepository) : ViewModel() {
     var _tourPointsList: MutableLiveData<List<TourPointPath>> = MutableLiveData()
     val tourPoints: LiveData<List<TourPointPath>> = _tourPointsList
     var originalList: List<TourPointPath> = listOf()
 
-    fun fetchTourPoints(context: Context) = viewModelScope.launch {
-        _tourPointsList.value = repository.getAllTourPoints(context)
+    fun fetchTourPoints(cityId: String) {
+        _tourPointsList.value = tourPointLocalDB.getAllTourPointsByCityId(cityId)
```

Comment:
better store it on a variable first so you don't have to force unwrap it like this

Reply:
I don't understand Jose, I'm not unwrapping I think. What do you mean?

Reply:
I already understood, I will fix it.

Reply:
Already fix it.